### PR TITLE
Update bamcmp

### PR DESCRIPTION
**Pull request by SPPearce**
Bamcmp module currently doesn't allow specification of the filenames for the two bam files it generates, nor does it check if there is a collision in these.

### CODE REVIEW

File: modules/nf-core/bamcmp/main.nf [hunk 1]
```diff
@@ -8,26 +8,37 @@ process BAMCMP {
...
+    path "versions.yml"                    , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
-    def prefix = task.ext.prefix ?: "${meta.id}"
+    prefix = task.ext.prefix ?: "${meta.id}_primary"
+    prefix2 = task.ext.prefix2 ?: "${meta.id}_contaminant"
```
Comment (nvnieuwk) [inline modules/nf-core/bamcmp/main.nf:24]: Seems controversial, but I like this solution :partying_face:

File: modules/nf-core/bamcmp/main.nf
Comment (nvnieuwk) [inline modules/nf-core/bamcmp/main.nf:1]: Can you add a stub section?

**Review by nvnieuwk — COMMENTED** [review]
One comment :p

File: modules/nf-core/bamcmp/main.nf
Comment (SPPearce) [reply modules/nf-core/bamcmp/main.nf:1]: stub added

**Review by nvnieuwk — APPROVED** [review]
LGTM

**Pull request merged**